panther_battery/test/test_dual_battery_publisher.cpp

```diff
@@ -0,0 +1,229 @@
+  }
+
+  uint8_t MergeBatteryPowerSupplyStatus(
+    const BatteryStateMsg & battery_msg_1, const BatteryStateMsg & battery_msg_2) const
+  {
+    return DualBatteryPublisher::MergeBatteryPowerSupplyStatus(battery_msg_1, battery_msg_2);
+  }
+
+  void MergeBatteryPowerSupplyHealth(
+    BatteryStateMsg & battery_msg, const BatteryStateMsg & battery_msg_1,
+    const BatteryStateMsg & battery_msg_2)
+  {
+    return DualBatteryPublisher::MergeBatteryPowerSupplyHealth(
+      battery_msg, battery_msg_1, battery_msg_2);
+  }
+};
+
+class TestDualBatteryPublisher : public testing::Test
+{
+public:
+  TestDualBatteryPublisher();
+  ~TestDualBatteryPublisher() {}
+
+protected:
+  rclcpp::Node::SharedPtr node_;
+  rclcpp::Subscription<BatteryStateMsg>::SharedPtr battery_sub_;
+  rclcpp::Subscription<BatteryStateMsg>::SharedPtr battery_1_sub_;
+  rclcpp::Subscription<BatteryStateMsg>::SharedPtr battery_2_sub_;
+
+  std::shared_ptr<panther_battery::Battery> battery_1_;
+  std::shared_ptr<panther_battery::Battery> battery_2_;
+  std::shared_ptr<DualBatteryPublisherWrapper> battery_publisher_;
+  BatteryStateMsg::SharedPtr battery_state_;
+  BatteryStateMsg::SharedPtr battery_1_state_;
+  BatteryStateMsg::SharedPtr battery_2_state_;
+};
+
+TestDualBatteryPublisher::TestDualBatteryPublisher()
+{
+  panther_battery::ADCBatteryParams params = {10, 10, 10, 10};
+  battery_1_ = std::make_shared<panther_battery::ADCBattery>(
+    [&]() { return 1.6; }, [&]() { return 0.02; }, [&]() { return 1.6; }, [&]() { return 0.4; },
+    params);
+  battery_2_ = std::make_shared<panther_battery::ADCBattery>(
+    [&]() { return 1.6; }, [&]() { return 0.02; }, [&]() { return 1.6; }, [&]() { return 0.4; },
+    params);
+
+  node_ = std::make_shared<rclcpp::Node>("node");
+  battery_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_state_ = msg; });
+  battery_1_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery_1_raw", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_1_state_ = msg; });
+  battery_2_sub_ = node_->create_subscription<BatteryStateMsg>(
+    "/battery_2_raw", 10, [&](const BatteryStateMsg::SharedPtr msg) { battery_2_state_ = msg; });
+  battery_publisher_ = std::make_shared<DualBatteryPublisherWrapper>(node_, battery_1_, battery_2_);
+}
+
+TEST_F(TestDualBatteryPublisher, CorrectTopicPublished)
+{
+  battery_publisher_->Publish();
+  ASSERT_TRUE(
+    panther_utils::test_utils::WaitForMsg(node_, battery_state_, std::chrono::milliseconds(1000)));
+  battery_publisher_->Publish();
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    node_, battery_1_state_, std::chrono::milliseconds(1000)));
+  battery_publisher_->Publish();
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    node_, battery_2_state_, std::chrono::milliseconds(1000)));
+}
+
+TEST_F(TestDualBatteryPublisher, MergeBatteryMsg)
```

Comment:
I think that this test should be moved after `MergeBatteryPowerSupplyStatus` and `MergeBatteryMsgHealth`